urllib3/poolmanager.py

```diff
@@ -227,8 +227,8 @@ def __init__(self, proxy_url, num_pools=10, headers=None,
             port = port_by_scheme.get(proxy.scheme, 80)
             proxy = proxy._replace(port=port)
 
-        assert proxy.scheme in ("http", "https"), \
-            'Not supported proxy scheme %s' % proxy.scheme
+        if proxy.scheme not in ("http", "https"):
+            raise ValueError('Not supported proxy scheme %s' % proxy.scheme)
```

Comment:
This is horribly backwards incompatible. `assert` raises an `AssertionError`, this raises a `ValueError`.